swri_roscpp/include/swri_roscpp/topic_service_client.h

```diff
@@ -159,16 +233,26 @@ class TopicServiceClientRaw
     response_ = message;
   }
 
-  rclcpp::Node* node_;
+  rclcpp::Node::SharedPtr node_;
 };  // class TopicServiceClientRaw
 
 template<class MReq>
-class TopicServiceClient: public TopicServiceClientRaw<typename MReq:: Request, typename MReq:: Response>
+class TopicServiceClient : public TopicServiceClientRaw<typename MReq:: Request, typename MReq:: Response>
 {
 public:
+  template<typename RepT = int64_t, typename RatioT = std::milli>
+  bool
+  wait_for_service(
+    std::chrono::duration<RepT, RatioT> timeout = std::chrono::duration(std::chrono::nanoseconds(1s)))
```

Comment:
This is the line causing a build error when using the default value. It should be `std::chrono::duration<RepT, RatioT>(...)`

Also, a duration of 1ns intentional? Do we not want to default to 1s or something? If you want the user to pass something in then remove the default argument.

Reply:
Okay, it looks like this is only failing on Foxy, but not Galactic. The duration isn't 1 ns, it's 1 second with a nanosecond precision.